### PR TITLE
DATA Wadi and SWaT datasets  

### PR DESCRIPTION
**Pull request by Jad-yehya**
Adds Wadi and SWaT datasets to the benchmark

### CODE REVIEW

File: datasets/swat.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+with safe_import_context() as import_ctx:
+    import pandas as pd
+
+
+class Dataset(BaseDataset):
+    name = "SWAT"
+
+    install_cmd = "conda"
+    requirements = ["pandas", "scikit-learn"]
```
Comment (tomMoral) [inline datasets/swat.py]: There is no need for sklearn no?
```suggestion
    requirements = ["pandas"]
```

File: datasets/swat.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    parameters = {
+        "debug": [False],
+        "n_splits": [5],
+        "validation_size": [0.2],
```
Comment (tomMoral) [inline datasets/swat.py]: unused parameters
```suggestion
```

File: datasets/swat.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    def get_data(self):
+        path = get_data_path(key="SWAT")
+
+        if not (path / "swat_train2.csv").exists():
+            raise FileNotFoundError(
+                "Train data not found. Please download the data "
+                f"from the official repository and place it in {path}"
```
Comment (tomMoral) [inline datasets/swat.py]: Add the URL to explain where to find the data? We should help the user here

File: datasets/swat.py (hunk 1 above)
Comment (tomMoral) [reply datasets/swat.py]: Also, as pandas is a dependency of `benchopt`, not necessary to have it here so you could even remove the full requirements.

File: datasets/wadi.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+from benchopt import BaseDataset, safe_import_context
+from benchopt.config import get_data_path
+
+with safe_import_context() as import_ctx:
+    import pandas as pd
+
+URL_XTRAIN = "./data/WADI/WADI_14days_new.csv"
+URL_XTEST = "./data/WADI/WADI_attackdataLABLE.csv"
```
Comment (tomMoral) [inline datasets/wadi.py]: Unused

File: datasets/wadi.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    name = "WADI"
+
+    install_cmd = "conda"
+    requirements = ["pandas"]
+
+    parameters = {
+        "debug": [False],
+        "n_splits": [5],
+        "validation_size": [0.2],
+    }
```
Comment (tomMoral) [inline datasets/wadi.py]:
```suggestion
    parameters = {
        "debug": [False],
    }
```

File: datasets/wadi.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+        # Extract the target
+        X_test.columns = X_test.columns.str.strip()
+        y_test = X_test["Attack LABLE (1:No Attack, -1:Attack)"].values
+        X_test.drop(
+            columns=todrop + [
+                     "Attack LABLE (1:No Attack, -1:Attack)"],
+            inplace=True
+        )
+        X_test.ffill(inplace=True)
```
Comment (tomMoral) [inline datasets/wadi.py:72]: why do you fill the missing values like this?

File: solvers/sktime_lof.py
Comment (tomMoral) [inline solvers/sktime_lof.py:1]: Not clear why this is in this PR?

File: solvers/sktime_lof.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    from sktime.annotation.lof import SubLOF
+    import pandas as pd
+    import numpy as np
+
+
+class Solver(BaseSolver):
+    name = "SubLOF"
+
+    install_cmd = "conda"
+    requirements = ["sktime", "pandas"]
```
Comment (tomMoral) [inline solvers/sktime_lof.py:14]:
```suggestion
    requirements = ["sktime"]
```

**Review by tomMoral — COMMENTED** [review]
A few comments

File: datasets/swat.py (hunk 1 above)
Comment (Jad-yehya) [reply datasets/swat.py]: True, I will remove it and check for the other references

File: datasets/wadi.py (hunk 6 above)
Comment (Jad-yehya) [reply datasets/wadi.py:72]: Here the only missing values are from the last 2 rows and values not missing repeat themselves, so I repeated the missing values

File: solvers/sktime_lof.py
Comment (Jad-yehya) [reply solvers/sktime_lof.py:1]: It should not be here, branch was not up to date with main

File: datasets/swat.py (hunk 3 above)
Comment (tomMoral) [reply datasets/swat.py]: Add a comment to explain how to get the data?

File: datasets/wadi.py (hunk 6 above)
Comment (tomMoral) [reply datasets/wadi.py:72]: Ok add a comment to explain this in the code.

File: datasets/swat.py (hunk 3 above)
Comment (Jad-yehya) [reply datasets/swat.py]: Yes, it will be clearer for the user

**Pull request merged**